lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -180,20 +171,139 @@ def scrub(transcript)
+        xml.transaction do
+          xml.send(transaction_type) do
+            xml.amount(amount(amount))
+            add_payment_source(xml, payment)
+            add_invoice(xml, options)
+          end
+        end
+      end
+
+      def cim_capture(amount, authorization, options)
+        transaction_id = split_authorization(authorization)[0]
+
+        commit(:cim_capture) do |xml|
+          add_order_id(xml, options)
+          xml.transaction do
+            xml.profileTransPriorAuthCapture do
+              xml.amount(amount(amount))
+              xml.transId(transaction_id)
+            end
+          end
+        end
+      end
+
+      def normal_capture(amount, authorization, options)
+        transaction_id = split_authorization(authorization)[0]
+
+        commit(:capture) do |xml|
+          add_order_id(xml, options)
+          xml.transactionRequest do
+            xml.transactionType('priorAuthCaptureTransaction')
+            xml.amount(amount(amount))
+            xml.refTransId(transaction_id)
+            add_invoice(xml, options)
+            add_user_fields(xml, amount, options)
+          end
+        end
+      end
+
+      def cim_refund(amount, authorization, options)
+        transaction_id, card_number = split_authorization(authorization)
+
+        commit(:cim_refund) do |xml|
+          add_order_id(xml, options)
+          xml.transaction do
+            xml.profileTransRefund do
+              xml.amount(amount(amount))
+              xml.creditCardNumberMasked(card_number)
+              add_invoice(xml, options)
+              xml.transId(transaction_id)
+            end
+          end
+        end
+      end
+
+      def normal_refund(amount, authorization, options)
+        transaction_id, card_number = split_authorization(authorization)
+
+        commit(:refund) do |xml|
+          xml.transactionRequest do
+            xml.transactionType('refundTransaction')
+            xml.amount(amount.nil? ? 0 : amount(amount))
+            xml.payment do
+              xml.creditCard do
+                xml.cardNumber(card_number || options[:card_number])
+                xml.expirationDate('XXXX')
+              end
+            end
+            xml.refTransId(transaction_id)
+
+            add_customer_data(xml, nil, options)
+            add_user_fields(xml, amount, options)
+          end
+        end
+      end
+
+      def cim_void(authorization, options)
+        commit(:cim_void) do |xml|
+          add_order_id(xml, options)
+          xml.transaction do
+            xml.profileTransVoid do
+              xml.transId(split_authorization(authorization)[0])
+            end
+          end
+        end
+      end
+
+      def normal_void(authorization, options)
+        commit(:void) do |xml|
+          add_order_id(xml, options)
+          xml.transactionRequest do
+            xml.transactionType('voidTransaction')
+            xml.refTransId(split_authorization(authorization)[0])
```

Comment:
Hmmm... not sure why it would be better to be before.

Talking to @ntalbott here, I'm gonna make a helper method for this `split` to be a bit more explicit.